src/plans/create/steps/storage-map/SourceStorageField.tsx

```diff
@@ -0,0 +1,88 @@
+import { type FC, useMemo } from 'react';
+import { Controller, useWatch } from 'react-hook-form';
+
+import FormGroupWithErrorText from '@components/common/FormGroupWithErrorText';
+import Select from '@components/common/MtvSelect';
+import { SelectGroup, SelectList, SelectOption } from '@patternfly/react-core';
+import { isEmpty } from '@utils/helpers';
+import { useForkliftTranslation } from '@utils/i18n';
+
+import { useCreatePlanFormContext } from '../../hooks';
+
+import { StorageMapFieldId, type StorageMapping } from './constants';
+
+type SourceStorageFieldProps = {
+  fieldId: string;
+  usedLabels: string[];
+  otherLabels: string[];
+};
+
+const SourceStorageField: FC<SourceStorageFieldProps> = ({ fieldId, otherLabels, usedLabels }) => {
+  const { control, trigger } = useCreatePlanFormContext();
+  const { t } = useForkliftTranslation();
+  const storageMappings = useWatch({ control, name: StorageMapFieldId.StorageMap });
+  const noStorageOption = useMemo(
+    () => <SelectOption isDisabled={true}>{t('No storages in this category')}</SelectOption>,
+    [t],
+  );
```

Comment:
this cans be extracted as a component and used also in network, with a type prop